lib/util/exec/hermit.ts

```diff
@@ -0,0 +1,90 @@
+import os from 'os';
+import fs from 'fs-extra';
+import upath from 'upath';
+import { GlobalConfig } from '../../config/global';
+import { logger } from '../../logger';
+import { rawExec } from './common';
+import type { RawExecOptions } from './types';
+
+export function isHermit(): boolean {
+  const { binarySource } = GlobalConfig.get();
+  return binarySource === 'hermit';
+}
+
+async function statFile(f: string): Promise<fs.Stats | undefined> {
+  let exists: fs.Stats | undefined = undefined;
+  try {
+    exists = await fs.stat(f);
+  } catch (e) {
+    // not doing anything when file not exists for errors
+  }
+
+  return exists;
+}
```

Comment:
We've got a utility function to help with this: https://github.com/renovatebot/renovate/blob/851d8f4f7a5218cac8da978af6aa73326c018faa/lib/util/fs/index.ts#L208-L218